### PR TITLE
azurelinux-rpm-macros: include release in elf module version

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Include the release information in the `version` field of the embedded ELF module metadata.
Before:
```
$ readelf --notes /usr/bin/tree | grep "description data" | sed -e "s/\s*description data: //g" -e "s
/ //g" | xxd -p -r | jq
{
  "moduleVersion": "1.8.0.0",
  "name": "tree",
  "os": "mariner",
  "osVersion": "3.0",
  "type": "rpm",
  "version": "1.8.0"
}
```
After:
```
$ readelf --notes /usr/bin/tree | grep "description data" | sed -e "s/\s*description data: //g" -e "s
/ //g" | xxd -p -r | jq
{
  "moduleVersion": "1.8.0.0",
  "name": "tree",
  "os": "mariner",
  "osVersion": "3.0",
  "type": "rpm",
  "version": "1.8.0-3.azl3"
}
```

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/53637270

###### Test Methodology

- Buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=639332&view=results
- Full build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=637663&view=results

### CODE REVIEW

File: SPECS/azurelinux-rpm-macros/gen-ld-script.sh [hunk 1]
```diff
@@ -6,26 +6,30 @@
 # gen-ld-script.sh
 # Generate linker script to embed ELF binaries with build metadata
 
-# /usr/lib/rpm/azl/gen-ld-script.sh %{name} %{version} %{_topdir} %{distro_release_version}
+# /usr/lib/rpm/azl/gen-ld-script.sh %{name} %{version}-%{release} %{_topdir} %{distro_release_version}
```
Comment (jslobodzian) [inline SPECS/azurelinux-rpm-macros/gen-ld-script.sh:9]: is epoch missing?

File: SPECS/azurelinux-rpm-macros/gen-ld-script.sh [hunk 2]
```diff
@@ -6,26 +6,30 @@
...
 OS_VERSION=$(echo $4 | cut -d. -f1,2)
 
+# When generating moduleVersion, strip everything after the dash.
+# ex: "1.8.0-2.azl3" -> "1.8.0",
+VERSION_NO_RELEASE=$(echo $2 | cut -d- -f1)
+
 # Count number of dot separators in $2 (version)
-NUM_DOT_SEPARATORS="${2//[^.]}"
+NUM_DOT_SEPARATORS="${VERSION_NO_RELEASE//[^.]}"
 # Ensure moduleVersion contains 4 version parts by adding ".0" padding
```
Comment (jslobodzian) [inline SPECS/azurelinux-rpm-macros/gen-ld-script.sh:20]: can you add a note that this padded dot stuff is required for Azure Watson

File: SPECS/azurelinux-rpm-macros/gen-ld-script.sh (hunk 1 above)
Comment (anphel31) [reply SPECS/azurelinux-rpm-macros/gen-ld-script.sh]: Yes, I'll add the epoch if it is defined for each package

File: SPECS/azurelinux-rpm-macros/gen-ld-script.sh (hunk 2 above)
Comment (anphel31) [reply SPECS/azurelinux-rpm-macros/gen-ld-script.sh]: Sure; updated

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**